src/lsm/level_data_iterator.zig

```diff
@@ -50,142 +56,113 @@ pub fn LevelDataIteratorType(comptime Table: type, comptime Storage: type) type
         context: Context,
 
         /// Internal state.
-        level_index_iterator: LevelIndexIterator,
         table_data_iterator: TableDataIterator,
+        table_index: usize = 0,
 
-        // Local copy of index block, for use in `table_data_iterator`.
-        index_block: BlockPtr,
+        read: Grid.Read = undefined,
+        next_tick: Grid.NextTick = undefined,
 
         callback: union(enum) {
             none,
             level_next: Callback,
             table_next: Callback,
         },
 
-        pub fn init(
-            allocator: mem.Allocator,
-        ) !LevelDataIterator {
-            var level_index_iterator = try LevelIndexIterator.init(allocator);
-            errdefer level_index_iterator.deinit(allocator);
-
-            var table_data_iterator = try TableDataIterator.init(allocator);
-            errdefer table_data_iterator.deinit(allocator);
+        pub fn init() !LevelIterator {
+            var table_data_iterator = try TableDataIterator.init();
+            errdefer table_data_iterator.deinit();
 
-            const index_block = try alloc_block(allocator);
-            errdefer allocator.free(index_block);
-
-            return LevelDataIterator{
+            return LevelIterator{
                 .context = undefined,
-                .level_index_iterator = level_index_iterator,
                 .table_data_iterator = table_data_iterator,
-                .index_block = index_block,
                 .callback = .none,
             };
         }
 
-        pub fn deinit(it: *LevelDataIterator, allocator: mem.Allocator) void {
-            allocator.free(it.index_block);
-            it.table_data_iterator.deinit(allocator);
-            it.level_index_iterator.deinit(allocator);
+        pub fn deinit(it: *LevelIterator) void {
+            it.table_data_iterator.deinit();
             it.* = undefined;
         }
 
         pub fn start(
-            it: *LevelDataIterator,
+            it: *LevelIterator,
             context: Context,
         ) void {
             assert(it.callback == .none);
```

Comment:
Does `assert(context.tables.len > 0);` not work for some reason?

Reply:
Ah, this is strange. I did reply to this thread but instead of commenting I believe I accidentally hit `Start Review`, so maybe you can't see my comment? It also says `Pending` on the comment. 🤦🏼 

Pasting the comment here:

I only added assert(context.level < constants.lsm_levels);. We can't assert assert(context.tables.len > 0); since we invoke LevelTableValueBlockIteratorType.start() for the case where the Immutable Table is being compacted into Level-0, even if there are no tables in Level 0 that intersect with the Immutable Table.

Reply:
I had to hit `Finish Review` to post all of the comments that I made 4 days ago, I believe they'd all visible to you now. 😅